src/Language/PureScript/Sugar/Names/Common.hs

```diff
@@ -24,14 +24,45 @@ warnDuplicateRefs
   -> m ()
 warnDuplicateRefs pos toError refs = do
   let withoutCtors = deleteCtors `map` refs
-      dupeRefs = mapMaybe (refToName pos) $ withoutCtors \\ nub withoutCtors
+      dupeRefs = mapMaybe (refToName pos) $ onlyDuplicates withoutCtors
       dupeCtors = concat $ mapMaybe (extractCtors pos) refs
 
   for_ (dupeRefs ++ dupeCtors) $ \(pos', name) ->
     warnWithPosition pos' . tell . errorMessage $ toError name
 
   where
 
+  -- Finds all duplicate refs
+  --onlyDuplicates :: forall a. Eq a => Ord a => [a] -> [a]
```

Comment:
I'd like to include type annotations for the inner functions, but encountering syntax errors with `forall a .`.